src/Content/Text/BBCode.php

```diff
@@ -636,4 +646,1341 @@ public static function limitBodySize($body)
+			} else {
+				$text = "[url=" . $match[2] . "]" . $match[2] . "[/url]";
+
+				// if its not a picture then look if its a page that contains a picture link
+				$body = Network::fetchUrl($match[1]);
+
+				$doc = new DOMDocument();
+				@$doc->loadHTML($body);
+				$xpath = new DomXPath($doc);
+				$list = $xpath->query("//meta[@name]");
+				foreach ($list as $node) {
+					$attr = [];
+
+					if ($node->attributes->length) {
+						foreach ($node->attributes as $attribute) {
+							$attr[$attribute->name] = $attribute->value;
+						}
+					}
+
+					if (strtolower($attr["name"]) == "twitter:image") {
+						$text = "[url=" . $attr["content"] . "]" . $attr["content"] . "[/url]";
+					}
+				}
+			}
+			Cache::set($match[1], $text);
+		}
+
+		return $text;
+	}
+
+	private static function expandLinksCallback($match)
+	{
+		if (($match[3] == "") || ($match[2] == $match[3]) || stristr($match[2], $match[3])) {
+			return ($match[1] . "[url]" . $match[2] . "[/url]");
+		} else {
+			return ($match[1] . $match[3] . " [url]" . $match[2] . "[/url]");
+		}
+	}
+
+	private static function cleanPictureLinksCallback($match)
+	{
+		$text = Cache::get($match[1]);
+
+		if (is_null($text)) {
+			$a = get_app();
+
+			$stamp1 = microtime(true);
+
+			$ch = @curl_init($match[1]);
+			@curl_setopt($ch, CURLOPT_NOBODY, true);
+			@curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+			@curl_setopt($ch, CURLOPT_USERAGENT, $a->get_useragent());
+			@curl_exec($ch);
+			$curl_info = @curl_getinfo($ch);
+
+			$a->save_timestamp($stamp1, "network");
+
+			// if its a link to a picture then embed this picture
+			if (substr($curl_info["content_type"], 0, 6) == "image/") {
+				$text = "[img]" . $match[1] . "[/img]";
+			} else {
+				$text = "[img]" . $match[2] . "[/img]";
+
+				// if its not a picture then look if its a page that contains a picture link
+				$body = Network::fetchUrl($match[1]);
+
+				$doc = new DOMDocument();
+				@$doc->loadHTML($body);
+				$xpath = new DomXPath($doc);
+				$list = $xpath->query("//meta[@name]");
+				foreach ($list as $node) {
+					$attr = [];
+					if ($node->attributes->length) {
+						foreach ($node->attributes as $attribute) {
+							$attr[$attribute->name] = $attribute->value;
+						}
+					}
+
+					if (strtolower($attr["name"]) == "twitter:image") {
+						$text = "[img]" . $attr["content"] . "[/img]";
+					}
+				}
+			}
+			Cache::set($match[1], $text);
+		}
+
+		return $text;
+	}
+
+	public static function cleanPictureLinks($text)
+	{
+		$return = preg_replace_callback("&\[url=([^\[\]]*)\]\[img\](.*)\[\/img\]\[\/url\]&Usi", 'self::cleanPictureLinksCallback', $text);
+		return $return;
+	}
+
+	private static function textHighlightCallback($match)
+	{
+		if (in_array(strtolower($match[1]),
+				['php', 'css', 'mysql', 'sql', 'abap', 'diff', 'html', 'perl', 'ruby',
+				'vbscript', 'avrc', 'dtd', 'java', 'xml', 'cpp', 'python', 'javascript', 'js', 'sh'])
+		) {
```

Comment:
This is the list of languages for which the syntax highlight works right? Could you add it to the BBCode documentation to the `[code]` tag reference.

Reply:
Done.